### ISSUE TITLE
MS Aaccess to PostgreSQL: PUBLIC vs public schema

### ISSUE THREAD

**Issue by BorisDomajnko** [keeps/dbptk-developer#282]
I have created a SIARD package from the MS Access database. When I restore a db in postgresql, the tool wants to create the table in the PUBLIC schema, but the default is public (already existing in PostgreSQL instalation). 

```
2016-11-18 07:55:38,992 [main] DEBUG (c.d.m.j.o.JDBCExportModule) Got a batch update exception while executing a batch statement
java.sql.BatchUpdateException: Batch entry 0 CREATE TABLE "PUBLIC"."PRS_DELEZ_SUBJ" ("ENOTA" numeric(30) NOT NULL, "ID" numeric(30) NOT NULL, "VISINA_DELEZA" varchar(250) NULL, "ZNESEK" numeric(18,2) NULL, "SIFRA_VALUTE" varchar(3) NULL, "DATUM_VPISA" timestamp without time zone NULL, "DATUM_IZBRISA" timestamp without time zone NULL) was aborted.  Call getNextException to see the cause.
	at org.postgresql.jdbc.BatchResultHandler.handleError(BatchResultHandler.java:133)
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2004)
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:360)
	at org.postgresql.jdbc.PgStatement.executeBatch(PgStatement.java:1019)
	at com.databasepreservation.modules.jdbc.out.JDBCExportModule.statementExecuteAndClearBatch(JDBCExportModule.java:802)
	at com.databasepreservation.modules.jdbc.out.JDBCExportModule.handleStructure(JDBCExportModule.java:301)
	at com.databasepreservation.modules.siard.in.input.SIARDImportDefault.getDatabase(SIARDImportDefault.java:46)
	at com.databasepreservation.Main.run(Main.java:131)
	at com.databasepreservation.Main.internal_main(Main.java:84)
	at com.databasepreservation.Main.main(Main.java:62)
2016-11-18 07:55:38,992 [main] DEBUG (c.d.m.j.o.JDBCExportModule) Next exception
org.postgresql.util.PSQLException: ERROR: schema "PUBLIC" does not exist
```

Steps required to reproduce the bug:

```
java -Xmx1200m -Dfile.encoding=UTF-8 -jar dbptk-app-2.0.0-beta6.jar -i microsoft-access -if d:\projekti\earh\databases\mdb\Prs1.accdb -e siard-2 -ef out_msa1.siard -etf out_msa1_tables.txt 

java -Xmx1000m -Dfile.encoding=UTF-8 -jar dbptk-app-2.0.0-beta6.jar -e postgresql -eh 10.0.92.38 -edb eark_test.msa1 -eu pg_eark -ep x -ede -epn 5432 -i siard-2 -if out_msa1.siard 
```

Attach the dbptk-app.log.txt file below.
[dbptk-app.log.txt](https://github.com/keeps/db-preservation-toolkit/files/599231/dbptk-app.log.txt)

**Comment by chalkos** [comment]
In this case, since the original schema was named PUBLIC, and Postgres supports multiple schemas, the DBPTK should try to create the schema PUBLIC and use that one. If the database had any other schema, that should also be created.

Is this the behaviour you were expecting?

**Comment by chalkos** [comment]
Is this blocking a pilot?

**Comment by BorisDomajnko** [comment]
Not for the official pilot. But currently a simple database cannot be restored (unless I create the schema with upper letters manually in advance).
PostgreSQL doc:
_...In the previous sections we created tables without specifying any schema names. By default, such tables (and other objects) are automatically put into a schema named "public". Every new database contains such a schema._

**Comment by chalkos** [comment]
SIARD functionality is not hardly covered by the DBMS documentation, and some aspects need to be handled in a special way.

When the **DBMS supports multiple schemas** (such as PostgreSQL), the actual schemas should be created.

When the **DBMS does not support multiple schemas** (such as MySQL), the information from all schemas should be merged **or, as an alternative,** a single schema should be selected and converted to the DBMS.

-------------

Could you answer my question in my other comment? (*"Is this the behaviour you were expecting?"*)

**Comment by BorisDomajnko** [comment]
Yes. If the target schema does not exist, dbptk should first create it before trying to configure the tables, now it just wanted to write into PUBLIC (and PUBLIC appeared as a result form export from MS Access).

**Comment by chalkos** [comment]
Ok. This also seems to me like the correct way to handle the situation.

**Comment by BorisDomajnko** [comment]
public is always created automatically by PG when I create a target database. But manually I can always add schema PUBLIC (CREATE SCHEMA "PUBLIC") and they can coexist.
Therefore I think it is a bug in DBPTK that it cannot create "PUBLIC". Or export MDB to SIARD should use some other schema name as default.
Regards, Boris
[dbptk-app.log.txt](https://github.com/keeps/db-preservation-toolkit/files/3579590/dbptk-app.log.txt)
.

**Comment by BorisDomajnko** [comment]
Closed with dbptk-app-2.9.3.jar
